.github/scripts/build.js

```diff
@@ -81,6 +106,19 @@ async function userHasWriteAccessToRepo({ github }, username, repoOwner, repoNam
   return userHasWriteAccess
 }
 
+function getRefIdForPr(prNumber) {
+	// Trailing newline is for compatibility with previous bash SHA calculation
+	return createShortHash(`refs/pull/${prNumber}/merge\n`);
+}
+function getRefIdForBranch(branchName) {
+	// Trailing newline is for compatibility with previous bash SHA calculation
```

Comment:
where are these functions used other than above (in new code)?

Wouldn't it be better to add the `\n` at the point where it's printed, rather than in this function? I makes this function less useful in the future

Reply:
These functions are used to generate the ID for the deployment resource group etc. In the current workflows they are generated via bash (`REFID=$(echo "${github_pr_ref}" | shasum | cut -c1-8)`).

These IDs are currently calculated based on the PR/branch and need to be consistent across the environment creation and destruction, so the newline is included to ensure consistency with the bash script approach so that when I add changes to use the JavaScript-generated IDs for creation, the destroy step still finds the environment

Reply:
Ah, right. Makes sense.